From ef7b3177adb3673a482c68cb62281cd9a5ca8e8d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?P=C3=A9ter=20Radics?= <mitchnull@gmail.com>
Date: Wed, 30 Dec 2015 16:58:15 +0100
Subject: [PATCH 32/34] Honor RememberLastUser and RememberLastSession
 settings.

Revert to default if RememberLastUser or RememberLastSession is
set to false.

Fix issue #527.
---
 src/daemon/Display.cpp | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/daemon/Display.cpp b/src/daemon/Display.cpp
index 7e35f3f..2e6e123 100644
--- a/src/daemon/Display.cpp
+++ b/src/daemon/Display.cpp
@@ -304,8 +304,14 @@ namespace SDDM {
             m_auth->setCookie(qobject_cast<XorgDisplayServer *>(m_displayServer)->cookie());
 
             // save last user and last session
-            stateConfig.Last.User.set(m_auth->user());
-            stateConfig.Last.Session.set(m_sessionName);
+            if (mainConfig.Users.RememberLastUser.get())
+                stateConfig.Last.User.set(m_auth->user());
+            else
+                stateConfig.Last.User.setDefault();
+            if (mainConfig.Users.RememberLastSession.get())
+                stateConfig.Last.Session.set(m_sessionName);
+            else
+                stateConfig.Last.Session.setDefault();
             stateConfig.save();
 
             if (m_socket)
-- 
2.5.0

